### DISCUSSION TITLE
The `returned` keyword

### DISCUSSION THREAD

**Discussion by jarzec** [carbon-language/carbon-lang#1889] [Language design]
I very much like the idea of the "explicit NRVO" proposed for Carbon. I am a big fan of "expicit over implicit". However, I personally find the syntax a bit verbose in simple cases. I think in such cases it could be replaced with the familar prameter declaration syntax for declaring a named return variable. 
To show better what I mean let me transform the example from [the design page](https://github.com/carbon-language/carbon-lang/tree/trunk/docs/design#returned-var):
```
fn MakeCircle(radius: i32) -> Circle {
  returned var c: Circle;
  c.radius = radius;
  // `return c` would be invalid because `returned` is in use.
  return var;
}
```
The proposed variant would be:
```
fn MakeCircle(radius: i32) -> c: Circle {
  c.radius = radius;
  return var;
}
```
I understood that the `returned var` syntax was intended to allow for limiting the scope of the returned variable. Nevertheless, I still think that in many simple cases just being able to name the return variable as shown above would make the code nicer to read.
The proposed variant would be just syntactic sugar for defining a `returned var` at function scope, so I don't think it would add any new issues.
What do you think? I am no programming language expert but I would be more than happy to help implement this proposition to learn how to contribute.

**Comment by geoffromer** [comment]
The original proposal discusses this alternative [here](https://github.com/carbon-language/carbon-lang/blob/trunk/proposals/p0257.md#named-return-variable-in-place-of-a-return-type).

**Reply by jarzec** [reply]
Thank you for pointing this out. I understand the reasoning. I just though of this more as a variant rather than alternative.
